response_operations_ui/controllers/notify_controller.py

```diff
@@ -39,14 +37,16 @@ def _send_message(self, email, template_id, personalisation=None, reference=None
             notification.update({"reference": reference})
 
         url = f'{self.notify_url}{template_id}'
-
         auth = app.config['SECURITY_USER_NAME'], app.config['SECURITY_USER_PASSWORD']
         response = requests.post(url, auth=auth, json=notification)
-        if response.status_code == 201:
+        status_code = response.status_code
+
+        if 201 <= status_code <= 399:
             logger.info('Notification id sent via Notify-Gateway to GOV.UK Notify.', id=response.json().get("id"))
         else:
             ref = reference if reference else 'reference_unknown'
-            raise NotifyError("There was a problem sending a notification to Notify-Gateway to GOV.UK Notify",
+            raise NotifyError(f'There was a problem sending a notification to Notify-Gateway. URL = {url},'
```

Comment:
As per @insacuri 's dismissed review, you don't need to pass these variables in as fstrings, `NotifyError` takes kwargs
```suggestion
            raise NotifyError("There was a problem sending a notification to Notify-Gateway.", url=url, status_code=status_code, message=response.text, reference=ref)
```